src/families/tron/AccountBalanceSummaryFooter.js

```diff
@@ -1,118 +1,130 @@
 // @flow
 
-import React from "react";
+import React, { useCallback, useMemo, useState } from "react";
 import { View, TouchableOpacity, ScrollView, StyleSheet } from "react-native";
-import { Trans } from "react-i18next";
-import { formatCurrencyUnit } from "@ledgerhq/live-common/lib/currencies";
+import { translate } from "react-i18next";
+import type { TFunction } from "react-i18next";
+import {
+  formatCurrencyUnit,
+  findCryptoCurrencyById,
+} from "@ledgerhq/live-common/lib/currencies";
+import { getCryptoCurrencyIcon } from "@ledgerhq/live-common/lib/reactNative";
+import type { CryptoCurrency } from "@ledgerhq/live-common/lib/types";
 import colors from "../../colors";
 import LText from "../../components/LText";
 import Info from "../../icons/Info";
+import InfoModal from "../../modals/Info";
+import type { ModalInfo } from "../../modals/Info";
+import FreezeIcon from "../../icons/Freeze";
+import BandwidthIcon from "../../icons/Bandwidth";
+import EnergyIcon from "../../icons/Energy";
 
-type Props = {
-  account: any,
-  countervalue: any,
-};
+interface Props {
+  account: any;
+  countervalue: any;
+  t: TFunction;
+}
 
 const formatConfig = {
   disableRounding: true,
   alwaysShowSign: false,
   showCode: true,
 };
 
-const AccountBalanceSummaryFooter = ({ account }: Props) => {
-  if (!account.tronResources) return null;
+type InfoName = "available" | "frozen" | "bandwidth" | "energy";
+
+function AccountBalanceSummaryFooter({ account, t }: Props) {
+  const [infoName, setInfoName] = useState<InfoName | typeof undefined>();
+  const infoCandidates = getInfoCandidates(t);
 
   const {
-    energy,
+    energy: formattedEnergy,
     bandwidth: { freeUsed, freeLimit, gainedUsed, gainedLimit } = {},
     tronPower,
   } = account.tronResources;
 
-  const spendableBalance = formatCurrencyUnit(
-    account.unit,
-    account.spendableBalance,
-    formatConfig,
+  const spendableBalance = useMemo(
+    () =>
+      formatCurrencyUnit(account.unit, account.spendableBalance, formatConfig),
+    [account.unit, account.spendableBalance],
+  );
+
+  const formattedBandwidth = useMemo(
+    () => freeLimit + gainedLimit - gainedUsed - freeUsed,
+    [freeLimit, gainedLimit, gainedUsed, freeUsed],
   );
 
-  const formatedEnergy = energy;
+  const onCloseModal = useCallback(() => {
+    setInfoName(undefined);
+  }, []);
+
+  const onPressInfoCreator = useCallback(
+    (infoName: InfoName) => () => setInfoName(infoName),
+    [],
+  );
+
+  const isModalOpen = useMemo(() => !!infoName, [infoName]);
+
+  const data = useMemo(() => (infoName ? infoCandidates[infoName] : []), [
```

Comment:
these memo are a bit overkill in my opinion you could just set data to `const data = infoCandidates[infoName] || [];`
and useMemo for a direct boolean value on isModalOpen also is a bit much :octocat:

Reply:
yup, I agree. I still have to stick with ternary operator due to its type tho.